Makefile

```diff
@@ -167,6 +167,20 @@ docker:
 .PHONY: build install update-tools tools godocs clean format lint \
 test-cli test-race test-unit test test-import
 
+test-contract:
+	env GOBIN= go get -u golang.org/x/tools/cmd/stringer
+	env GOBIN= go get -u github.com/kevinburke/go-bindata/go-bindata
+	env GOBIN= go get -u github.com/fjl/gencodec
+	env GOBIN= go get -u github.com/golang/protobuf/protoc-gen-go
+	env GOBIN= go install ./cmd/abigen
+	@type "npm" 2> /dev/null || echo 'Please install node.js and npm'
+	@type "solc" 2> /dev/null || echo 'Please install solc'
+	@type "protoc" 2> /dev/null || echo 'Please install protoc'
```

Comment:
I don't think it exits if the user hasn't installed the tools?

Reply:
correct... these will need to be installed via package manager (apt-get, apk, etc.) any tips? i thought that this would just prompt user to install these using their desired package manager. 
I have a compatible dokcker file for testing this internally as these tools arent necessary for running the client. Any comments on this?

Reply:
maybe use `exit 1` after each `echo`?

Reply:
i did that, you are a literal mind reader